main.js

```diff
@@ -1,26 +1,80 @@
-import oDate from './foundation/date';
-// TODO: integrate n-layout into this repo
-import layout from 'n-layout';
 // TODO: integrate next-js-setup into this repo
 import setup from 'next-js-setup';
-// TODO: these should be configurable as they're not foundations
-import prompts from './components/message-prompts';
+import date from './foundation/date';
+import header from './components/header';
+import cookieMessage from './components/cookie-message';
+import welcomeMessage from './components/welcome-message';
+import messagePrompts from './components/message-prompts';
 import { client as myFtClient, ui as myFtUi } from './components/myft';
 
+const presets = {
+	discrete: {
+		header: true,
+		date: true
+	},
+	all: {
+		header: true,
+		date: true,
+		cookieMessage: true,
+		welcomeMessage: true,
+		myft: true,
+		messagePrompts: true
+	}
+}
+
 module.exports = {
-	bootstrap: function (cb, opts = {}) {
+	bootstrap: function (opts = {}, cb) {
+
+		if (opts.preset) {
+			opts = Object.assign(presets[opts.preset], opts);
+		}
+
 		return setup.bootstrap(({ flags }) => {
-			// NOTE: make sure we init myft client *before* n-layout
-			const clientOpts = [];
-			flags.get('follow') && clientOpts.push({relationship: 'followed', type: 'concept'});
-			flags.get('saveForLater') && clientOpts.push({relationship: 'saved', type: 'content'});
-			const myftClient = myFtClient.init(clientOpts);
-
-			layout.init(flags, opts);
-			oDate.init();
-			prompts.init();
-			myFtUi.init({ anonymous: !(/FTSession=/.test(document.cookie)) });
-			return Promise.resolve({flags}).then(cb);
+
+			if (opts.myft) {
```

Comment:
I wonder how we can do this so that these things are not bundled? We have the environment vars plugin in webpack which helps remove dead code but not sure if it prunes whole dependencies.